sdcm/utils/_alternator/api.py

```diff
@@ -0,0 +1,165 @@
+        return self.alternator_apis[endpoint_url]
+
+    def set_write_isolation(self, isolation, endpoint_url=None, table_name=consts.TABLE_NAME):
+        dynamodb_api = self.get_dynamodb_api(endpoint_url=endpoint_url)
+        isolation = isolation if not isinstance(isolation, enums.WriteIsolation) else isolation.value
+        got = dynamodb_api.client.describe_table(TableName=table_name)['Table']
+        arn = got['TableArn']
+        tags = [
+            {
+                'Key': 'system:write_isolation',
+                'Value': isolation
+            }
+        ]
+        dynamodb_api.client.tag_resource(ResourceArn=arn, Tags=tags)
+
+    def create_table(self, endpoint_url=None,  # pylint: disable=too-many-arguments
+                     schema=enums.YCSVSchemaTypes.HASH_AND_RANGE, isolation=None, table_name=consts.TABLE_NAME,
+                     wait_until_table_exists=True, **kwargs) -> Table:
+        if isinstance(schema, enums.YCSVSchemaTypes):
+            schema = schema.value
+        schema = schemas.ALTERNATOR_SCHEMAS[schema]
+        dynamodb_api = self.get_dynamodb_api(endpoint_url=endpoint_url)
+        LOGGER.debug(f"Creating a new table '{table_name}' using endpoint '{endpoint_url}'")
+        table = dynamodb_api.resource.create_table(
+            TableName=table_name, BillingMode="PAY_PER_REQUEST", **schema, **kwargs)
+        if wait_until_table_exists:
+            waiter = dynamodb_api.client.get_waiter('table_exists')
+            waiter.wait(TableName=table_name, WaiterConfig=dict(Delay=1, MaxAttempts=100))
+
+        LOGGER.info(f"The table '{table_name}' successfully created..")
+        response = dynamodb_api.client.describe_table(TableName=table_name)
+
+        if isolation:
+            self.set_write_isolation(isolation=isolation, endpoint_url=endpoint_url, table_name=table_name)
+        LOGGER.debug(f"Table's schema and configuration are: {response}")
+        return table
+
+    def scan_table(self, endpoint_url=None, table_name=consts.TABLE_NAME, threads_num=None, **kwargs):
+        scan_result, is_parallel_scan = [], threads_num and threads_num > 0
+        dynamodb_api = self.get_dynamodb_api(endpoint_url=endpoint_url)
+        table = dynamodb_api.resource.Table(name=table_name)
+
+        def _scan_table(part_scan_idx=None):
+            parallel_params, result, still_running_while = {}, [], True
+            if is_parallel_scan:
+                parallel_params = {"TotalSegments": threads_num, "Segment": part_scan_idx}
+                LOGGER.debug(f"Starting parallel scan part '{part_scan_idx + 1}' on table '{table_name}'")
+            else:
+                LOGGER.debug(f"Starting full scan on table '{table_name}'")
+            while still_running_while:
+                response = table.scan(**parallel_params, **kwargs)
+                result.extend(response["Items"])
+                still_running_while = 'LastEvaluatedKey' in response
```

Comment:
Please try adding "Limit=1" and see if it really works well. I don't understand how it can work - you just repeat exactly the same request again and again, with no change! (or, am I missing something?) the "ExclusiveStartKey" parameter is supposed to change between iterations.

Reply:
fix